### PR TITLE
added mraa_boards.c and mraa_boards.h issue #642

### PR DESCRIPTION
**Pull request by nicolalunghiemutex**
Signed-off-by: Nicola Lunghi <nicola.lunghi@emutex.com>

### CODE REVIEW

**Comment by nicolalunghiemutex** [comment]
issue #642

**Comment by nicolalunghiemutex** [comment]
I've updated the branch please check

also there was an issue with up board 
the maximum length of a string now is 12 bit

the up code was

#define MAX_LENGTH 12

  strncpy(pin_info->name, name, MAX_LENGTH);

so it was copying only the first 8 byte on the string resulting in: -> not correctly copying the null terminator on string with more than 8 char

I2C1_SCL    #��|
UART1_RXion/����
SPI_MOSI
   ��
SPI_MISO    �-�	
I2S_DOUTmraa

File: include/mraa_boards.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   Example:
+
+   mraa_board_t* b = (mraa_board_t*) calloc(1, sizeof(mraa_board_t));
+   ...
+   b->pins = (mraa_pininfo_t*) malloc(sizeof(mraa_pininfo_t) * MRAA_UP_PINCOUNT);
+   ...
+   mraa_set_pininfo(b, 8, "UART1_TX",  (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 1 }, 14);
+*/
+mraa_result_t mraa_set_pininfo(mraa_board_t* board, int mraa_index, char* name,
+							   mraa_pincapabilities_t caps, int sysfs_pin);
```
Comment (alext-mkrs) [inline include/mraa_boards.h]: This indentation is a bit off due to tabs, looks like - let's align to the opening parenthesis.

File: src/mraa_boards.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <string.h>
+#include <ctype.h>
+#include "mraa_boards.h"
+
+// utility function to setup pin mapping of boards
+mraa_result_t
+mraa_set_pininfo(mraa_board_t* board, int mraa_index, char* name,
+        mraa_pincapabilities_t caps, int sysfs_pin)
```
Comment (alext-mkrs) [inline src/mraa_boards.c]: The indentation is a bit off, let's align to the opening parenthesis.

File: src/mraa_boards.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        mraa_pininfo_t* pin_info = &board->pins[mraa_index];
+        strncpy(pin_info->name, name, MRAA_PIN_NAME_SIZE);
+        pin_info->capabilities = caps;
+        if (caps.gpio) {
+            pin_info->gpio.pinmap = sysfs_pin;
+            pin_info->gpio.mux_total = 0;
+        }
+        if (caps.pwm) {
+            int controller = 0;
+            if (strncmp(name, "PWM", 3) == 0 && strlen(name) > 3 && isdigit(name[3]))
```
Comment (alext-mkrs) [inline src/mraa_boards.c]: This assumes very specific PWM pin naming format - probably worth adding to documents or at least function description in the header?

Comment (alext-mkrs) [inline src/mraa_boards.c]: Even one-line blocks should have braces around per mraa's code style

**Pull request closed without merging**